### PR TITLE
CSB-3925: Request to offer connection pooling in camel-cics

### PR DESCRIPTION
**Pull request by luigidemasi**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by grgrzybek** [comment]
+1 and big admiration for what you've done here @luigidemasi

**Review by grgrzybek — APPROVED** [review]

File: camel-cics/pom.xml [hunk 1]
```diff
@@ -75,6 +75,19 @@
             <version>${slf4j-api-version}</version>
         </dependency>
 
+        <!-- CTG v.9.2 -->
+        <dependency>
```
Comment (Croway) [inline camel-cics/pom.xml]: isn't it duplicated from line 65?

Moreover, should it have scope provided? like we did for SAP? (just wondering)

File: camel-cics/src/test/java/com/redhat/camel/component/cics/AbstractCICSTest.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+public abstract class AbstractCICSTest extends CamelTestSupport {
+
+    public static final Logger LOG = LoggerFactory.getLogger(AbstractCICSTest.class);
+
+    protected final static Network network = Network.newNetwork();
+    /** CTG Server */
+    protected static final String CTG_SERVER = "cics_server_name";
+
+    protected static final GenericContainer<?> ctgContainer = new GenericContainer<>("images.paas.redhat.com/fuseqe/ibm-cicstg-container-linux-x86-trial:9.3")
```
Comment (Croway) [inline camel-cics/src/test/java/com/redhat/camel/component/cics/AbstractCICSTest.java:36]: the version used here is 9.3, while the ctg client lib is 9.2, is it ok?

**Review by Croway — APPROVED** [review]

File: camel-cics/src/test/java/com/redhat/camel/component/cics/AbstractCICSTest.java (hunk 2 above)
Comment (luigidemasi) [reply camel-cics/src/test/java/com/redhat/camel/component/cics/AbstractCICSTest.java:36]: We don't have choice, there's no 9.2 container, IBM started doing it from 9.3

**Pull request merged**